README.md

```diff
@@ -251,6 +251,21 @@ for await (const message of await conversation.streamMessages()) {
 }
 ```
 
+#### Listen for new messages in all conversations
+
+To listen for any new incoming/outgoing messages from *all* conversations, use `streamAllMessages()`:
```

Comment:
```suggestion
To listen for any new messages from *all* conversations, use `converstaions.streamAllMessages()`.

Note: there is a chance this stream can miss messages if multiple conversations begin at nearly the same time while we are updating the stream to account for a new conversation.
```

Reply:
Thank you so much for taking such initiative in opening this PR, @joshuasorkin!

And thank you to @elisealix22 for your refinements and the very important note!

Just a couple of nit tweaks and a question! <3

- Change`converstaions` --> `conversations`

- Change `Note: there` -> `**Note:** There`

- Question: In the statement `we are updating` - I wonder if it might be possible to say what "we" is? What is updating the stream? The client?

Reply:
@jhaaaa wow, thank you so much! I updated the suggestion to account for those notes.

@joshuasorkin it's actually due to the `contentTopicUpdater` which is called when a new message is received. So we start the stream listening for all existing conversations' content topics (where one content topic represents one existing conversation) as well as the invite topic to ensure we also receive new messages for new conversations as they are created.

The tricky part is once we receive an invite topic for a new conversation, we want to update the existing stream's content topics to include that new conversation's content topic so the stream can receive new messages for that conversation as well. To accomplish this, we use the `contentTopicUpdater` to update the stream's list of content topics as we receive messages indicating a new conversation has started. The potential race condition is that if we receive multiple new conversations in the time it takes to update the stream's content topics (I have yet to see this in practice for what it is worth), we could potentially drop a content topic while the stream is resubscribing to a different conversations' content topic.

https://github.com/xmtp/xmtp-js/blob/2868899477e1dc6afddc38e71214354c8e154b0b/src/conversations/Conversations.ts#L186

Reply:
@elisealix22 thank you for clarifying!  Is there a standard tag that I could drop into the comment to mark this as a theoretical bug?  I'd like it to show up on a search for issues that need to be addressed eventually but are not currently affecting performance.